src/engine/strat_engine/blockdevmgr.rs

```diff
@@ -299,16 +303,18 @@ pub fn initialize(pool_uuid: PoolUuid,
                         add_devs.push((dev, (devnode, dev_size, f)))
                     }
                 }
-                DevOwnership::Ours(uuid) => {
-                    if pool_uuid != uuid {
+                DevOwnership::Ours(uuid, dev_uuid) => {
+                    if pool_uuid == uuid {
+                        if !owned_devs.contains(&dev_uuid) {
+                            let error_str = format!("Device {} with pool UUID is unknown to pool",
+                                                    devnode.display());
+                            return Err(EngineError::Engine(ErrorEnum::Invalid, error_str));
```

Comment:
could eliminate error_str here.

Reply:
I'ld rather keep it that way because I find it easier to read. I think it might be so deeply indented that rustfmt couldn't handle it, but that is secondary.